content/en/blog/_posts/2020-03-30-two-phased-canary-rollout-with-gloo.md

```diff
@@ -0,0 +1,772 @@
+---
+title: " Two-phased Canary Rollout with Open Source Gloo "
+date: 2020-03-30
+slug: two-phased-canary-rollout-with-gloo
+url: /blog/2020/03/Two-phased-Canary-Rollout-With-Gloo
+---
+
+Every day at [Solo](https://www.solo.io/), we're talking to platform owners, architects, and engineers who are using [Gloo](https://www.solo.io/products/gloo/) as an API gateway
+to expose their applications to end users. These applications may span legacy monoliths, microservices, managed cloud services, and Kubernetes 
+clusters. Fortunately, **Gloo makes it easy to set up routes to manage, secure, and observe application traffic while 
+supporting a flexible deployment architecture** to meet our customers' varying production needs.
+
+Beyond the initial set up, platform owners frequently ask us to help design the **operational workflows** within their organization: 
+How do we bring a new application online? How do we upgrade an application? How do we divide responsibilities across our 
+platform, ops, and development teams? 
+
+In this post, we're going to use Gloo to design a **two-phased canary rollout** workflow for application upgrades:
+* In the first phase, we'll do canary testing by shifting a small subset of traffic to the new version. This allows you to safely perform smoke and correctness tests. 
+* In the second phase, we'll progressively shift traffic to the new version, allowing us to monitor the new version under load, and eventually, decommission the old version. 
+
+To keep it simple, we're going to focus on designing the workflow using [open source Gloo](https://github.com/solo-io/gloo), and we're going to deploy the gateway and 
+application to Kubernetes. At the end, we'll talk about a few extensions and advanced topics that could be interesting to explore in a follow up.  
+
+## Initial setup
+
+To start, we need a Kubernetes cluster. This example doesn't take advantage of any cloud specific 
+features, and can be run against a local test cluster such as [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/). 
+This post assumes a basic understanding of Kubernetes and how to interact with it using `kubectl`. 
+
+We'll install the latest [open source Gloo](https://github.com/solo-io/gloo) to the `gloo-system` namespace and deploy 
+ version `v1` of an example application to the `echo` namespace. We'll expose this application outside the cluster 
+ by creating a route in Gloo, to end up with a picture like this: 
+
+![](https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/1-setup/setup.png)
```

Comment:
1. This image should have an alt attribute.
2. Can you copy the image into the repo so that it's served via the content CDN and the image is in source control?